src/lib/zAuction.ts

```diff
@@ -87,6 +87,39 @@ export async function getBidsForNft(
 	return bids;
 }
 
+export async function getBidsListForNfts(
+	baseApiUri: string,
+	contract: string[],
+	tokenId: string[],
+) {
+	const nftIdsArray: string[] = [];
+	const bidsList: string[] = [];
+
+	for (let i = 0; i < tokenId.length; i++) {
+		// it builds the array of nftIds strings
+		nftIdsArray.push(getNftId(contract[i], tokenId[i]));
+	}
+
+	const nftIdsBody = { nftIds: nftIdsArray }; //builds the correct body to send
+
+	let endpoints = getApiEndpoints(baseApiUri);
+
+	const response = await fetch(endpoints.bidListEndpoint, {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify(nftIdsBody),
+	});
+
+	const jsonResponse = await response.json(); //awaits the json of the response
```

Comment:
You should create an interface and type this to that interface. Currently `jsonResponse` is `any` which we want to avoid.